Allow rename files in metadata file store

josegar74 opened the pull request:
This change request allows to rename files in the datastore:
* move file on the filesystem
* updating the references in the metadata
* updating the database `metadatafileuploads`

[edit-resource-name.webm](https://github.com/user-attachments/assets/1ff76b74-256e-4b19-b8ec-ae8299f9c8a8)


If filename is already used, error is reported

<img width="372" height="201" alt="image" src="https://github.com/user-attachments/assets/e23be14d-7c6b-41ad-8ab2-2bde5fd67591" />





# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geonetwork/core-geonetwork/blob/main/CONTRIBUTING.md)
- [X] *Pull request* provided for `main` branch, backports managed with label
- [ ] *Good housekeeping* of code, cleaning up comments, tests, and documentation
- [X] *Clean commit history* broken into understandable chucks, avoiding big commits with hundreds of files, cautious of reformatting and whitespace changes
- [X] *Clean commit message*s, longer verbose messages are encouraged
- [ ] *API Changes* are identified in commit messages
- [ ] *Testing* provided for features or enhancements using [automatic tests](https://github.com/geonetwork/core-geonetwork/blob/main/software_development/TESTING.md)
- [ ] *User documentation* provided for new features or enhancements in [manual](https://github.com/geonetwork/core-geonetwork/tree/main/docs/manual)
- [ ] *Build documentation* provided for development instructions in `README.md` files
- [ ] *Library management* using `pom.xml` dependency management. Update build documentation with intended library use and library tutorials or documentation

Funded by: EEA

josegar74 committed:
Allow rename files in metadata file store

josegar74 committed:
Use PATCH endpoint to rename resource name

josegar74 committed:
Update metadata attachment reference when renaming a resource

PascalLike committed:
Fix compilation errors after rebasing on main

PascalLike committed:
Adding unsupported error for other storage backends

PascalLike committed:
Adding abstract and S3 renameResource

PascalLike committed:
Adding test

fxprunayre commented on core/src/main/java/org/fao/geonet/api/records/attachments/ResourceLoggerStore.java:
```diff
@@ -294,4 +294,13 @@ public void copyResources(ServiceContext context, String sourceUuid, String targ
     }
+
+    @Override
+    public MetadataResource renameResource(ServiceContext context, String metadataUuid, String resourceId, String newName, Boolean approved) throws Exception {
```
Here we need to update the filename in the database table.

fxprunayre commented on web-ui/src/main/resources/catalog/components/filestore/partials/filestore.html:
```diff
@@ -38,15 +38,46 @@
+                <span class="fa fa-cancel"></span>
+              </a>
+
+              <div data-ng-show="duplicatedFilename" data-translate="">
```
```suggestion
              <div data-ng-show="duplicatedFilename" data-translate="" class="alert alert-danger">
```

Display alert with `danger` class?

fxprunayre commented on web-ui/src/main/resources/catalog/components/filestore/FileStoreDirective.js:
```diff
@@ -346,7 +346,63 @@
+                      // Refresh the onlinesrc service to update the resource list and the metadata form
+                      // with the new resource name.
+                      gnOnlinesrc.refresh();
+                    });
```
Here we should report error if any eg. file name too long

PascalLike replied:
Set to 255, but probably we could set even lower.

fxprunayre replied:
While adding folder support, we will probably increase it so we can keep 255 for now.

PascalLike committed:
Fixes for sonarcloud

PascalLike committed:
update the filename in the database table

PascalLike committed:
changing style of duplicate filename to danger

PascalLike committed:
Adding a check on filename length max 255

PascalLike committed:
Limit filename length field

PascalLike committed:
Check error message

PascalLike committed:
Rename on METADATAFILEUPLOADS logic

PascalLike committed:
Adding default values

PascalLike commented:
There are some Sonarqube Cloud findings that I'm not sure need to be addressed (they may require changes beyond the scope of this pull request). Let me know what you think.

josegar74 committed:
Code fixes / improvements:

- FilesystemStore.renameResource preserve resource visibility
- S3Store.renameResource checks existint resource  before copying and throws ResourceAlreadyExistException
- AttachmentsApi change replaceAll to replace to avoid regex. Update @Parameter newResourceName.
- ResourceLoggerStore catch EmptyResultDataAccessException instead of Exception
- AbstractStore.renameResource rethrows SecurityException, IllegalArgumentException, ResourceNotFoundException, and ResourceAlreadyExistException as-is, so their correct HTTP status is preserved
- Add missing locale key renameResource

josegar74 committed:
Cancel edit resource name: unset editing resource flag
